app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/PlantCreationFragment.kt

```diff
@@ -118,6 +107,61 @@ class PlantCreationFragment : Fragment(R.layout.fragment_plant_creation) {
         }
     }
 
+    private fun setUpFrequencyViews(valuesList: List<Int>) {
+        val wateringValueList = valuesList.map {
+            resources.getQuantityString(
+                R.plurals.msg_creation_frequency_units, it, it
+            )
+        }
+
+        val frequencyAdapter = ArrayAdapter(
+            requireContext(),
+            R.layout.support_simple_spinner_dropdown_item,
+            wateringValueList
+        )
+
+        binding.etCreationWateringFrequency.setUpWithAdapter(
+            frequencyAdapter,
+            valuesList,
+            creationViewModel::onWateringFrequencySelected
+        )
+
+        binding.etCreationSprayingFrequency.setUpWithAdapter(
+            frequencyAdapter,
+            valuesList,
+            creationViewModel::onSprayingFrequencySelected
+        )
+
+        binding.etCreationLooseningFrequency.setUpWithAdapter(
+            frequencyAdapter,
+            valuesList,
+            creationViewModel::onLooseningFrequencySelected
+        )
+    }
+
+    private fun <T> AutoCompleteTextView.setUpWithAdapter(
```

Comment:
1. I think you can move this to seom Ext file
2. I thing Generics should be for ValuesList and onItemClick, and ArrayAdapter could be String